### ISSUE TITLE
Command creation multi-threading

### ISSUE THREAD

**Issue by prodigysml** [codingo/Interlace#75]
As [@br-sz](https://github.com/br-sn) raised, Interlace does not multi-thread command creation and substitution. This causes Interlace to hang for large amounts of time, when kicking off the tool. 

My suggestion is to use multi-threading on the variable replacement and creation of commands to speed up the process and use resources available efficiently.

**Comment by iamOgunyinka** [comment]
**tl;dr** This is not an easy decision to make and would involve a little bit of profiling of the tool to know where to engage. 

**Longer explanation**
> My suggestion is to use multi-threading on the...creation of commands to speed up the process.

Command creation comes from the sequential reading of the command file, and while this can be multithreaded, it adds another level of complexity which may even make performance worse in the long run. Reading data from a file is fastest when you access the file sequentially. That minimizes the number of reader head seeks(on HDDs), by far the most expensive operation on a disk drive. By splitting the reading across multiple threads, each reading a different part of the file, you are making the reader head constantly jump back and forth. Very, very bad for throughput.

> ...use multi-threading on the variable replacement

I believe this can be done. A little hiccup may only occur when a substitution for a command needs to be made into the command list. Take for example, the host and command list given below.
`command_list.txt`
```
echo _target_
echo "Starting scans"
scan _target_:_port_
```
`hosts.txt`
```
localhost
other_host
```
and the `_port_` is given as `80-82`
The resulting command should be similar to this
```
echo localhost
echo other_host
echo "Starting scans"
scan localhost:80
scan localhost:81
scan localhost:82
scan other_host:80
scan other_host:81
scan other_host:82
```
Now, assuming, a thread takes the unprocessed command list and starts processing variable `_host_` and another thread processes the `_port_` variable; combining the two lists correctly to give the above final command lists after being processed independently **may** add another problem and one has to answer questions similar to "which of these commands comes before?". Then there's the CPython's GIL issue(which I'm not particularly sure but would like to confirm if it) affects CPU-bound operations.

**Comment by codingo** [comment]
Closed with #73
